modules/planning/tasks/deciders/lane_change_decider/lane_change_decider.cc

```diff
@@ -188,37 +192,48 @@ void LaneChangeDecider::UpdateStatus(double timestamp,
 }
 
 void LaneChangeDecider::PrioritizeChangeLane(
+    const bool is_prioritize_change_lane,
     std::list<ReferenceLineInfo>* reference_line_info) const {
   if (reference_line_info->empty()) {
     AERROR << "Reference line info empty";
     return;
   }
+
   auto iter = reference_line_info->begin();
   while (iter != reference_line_info->end()) {
-    if (iter->IsChangeLanePath()) {
-      reference_line_info->splice(reference_line_info->begin(),
-                                  *reference_line_info, iter);
+    ADEBUG << "iter->IsChangeLanePath(): " << iter->IsChangeLanePath();
+    /* is_prioritize_change_lane == true: prioritize change_lane_reference_line
+       is_prioritize_change_lane == false: prioritize
+       non_change_lane_reference_line */
+    if (is_prioritize_change_lane == iter->IsChangeLanePath()) {
```

Comment:
The logic seems to work. However, I would not recommend writing comparison in this way. 'is_prioritize_change_lane' and 'iter->IsChangeLanePath()' represents totally different meanings. It's more clear to compare them separately with 'True' and 'False' in your logic.

Reply:
Fixed